### ISSUE TITLE
Add time tracking to Cards

### ISSUE THREAD

**Issue by PanCakeConnaisseur** [nextcloud/deck#2299]
# User Story
As a user of the deck app I would like to be able to see how much time I have spend on a task represented by a card. After completing a project I want to see how much time I spent in total and on each sub task. I want to see whether I maintained an allotted time budget and to be able to report it.

# Implementation Proposal
## Record Button
I think this could be implemented with a button on each card. The button could follow the common "record" symbol and when clicked would start a counter for this card. On start the button changes to a "stop" symbol (square). Alternatively the buttons can simply be called "Start Tracking Time" and "Stop Tracking Time".
## Time display
Each card has a rectangular area displaying a time in hh:mm:ss format. It is greyed out when no time has been record. The time starts counting when a user presses the record button and stops when the user stops counting. It continues running when a user starts the button again.
## Statistics
There is a cumulative display for each project that displays the sum of all cards.

# Optional Features
* to correct time in case it was not stopped by mistake
    * allow manual editing of time by clicking on it OR
    * show "details" button where user can see a list of all recorded snippets and edit them, e.g. deleting the last recordin

**Comment by PanCakeConnaisseur** [comment]
I am thinking about implementing this. Has anyone else done anything in this regards or would be willing to join forces?

**Comment by putt1ck** [comment]
We have a customer interested in a time tracking app in Nextcloud, but not Deck specific

**Comment by juliushaertl** [comment]
https://github.com/nextcloud/tasks/issues/75 is the related issue from the tasks app. Since this is quite a huge feature with lots of open questions both technically as well as from a user interaction perspective, I'd definitely say that the general implementation idea should be somehow sorted out before actually writing some code.

**Comment by juliushaertl** [comment]
Might be interesting to integrate with the new https://github.com/te-online/timemanager app by @te-online. It would probably be nice if there would be a way to easily create time tracking entries that are somehow associated to a specific task and on the deck app we could offer some API so that the time tracking app would be able to list the task related times right within the cards.

**Comment by thomasfrobieter** [comment]
> Might be interesting to integrate with the new https://github.com/te-online/timemanager app by @te-online. It would probably be nice if there would be a way to easily create time tracking entries that are somehow associated to a specific task and on the deck app we could offer some API so that the time tracking app would be able to list the task related times right within the cards.

Exactly, i've tested https://apps.nextcloud.com/apps/timetracker & https://apps.nextcloud.com/apps/timemanager both built their own project / task structure, sadly both without support for the 'core' tasks app.

**Comment by ogghi** [comment]
Hi there, digging up something old...is there any way to have time tracking in Deck?
My wife asked for this actually..she is used to Jira from the office, and thinks it would be nice to have at home as well for some things...

**Comment by xdubx** [comment]
Ping from 2024 :)

**Comment by PeGa** [comment]
Another one! I just learned that Planka has the perfect feature: https://raw.githubusercontent.com/plankanban/planka/master/demo.gif

Honestly I'd love to see something like that working in deck.

**Comment by xdubx** [comment]
> Another one! I just learned that Planka has the perfect feature: https://raw.githubusercontent.com/plankanban/planka/master/demo.gif
> 
> Honestly I'd love to see something like that working in deck.

copy code. Paste it here :D

**Comment by underclockeddev** [comment]
Throwing in a comment here just to let the devs know someone else would appreciate/use this feature too. Thanks for everything you do open-source devs! Loving Deck and getting a lot of value from it. Time-tracking on cards would make this super competitive to Jira for small team/solo projects.

**Comment by coolidfrez** [comment]
I work for the Sorbonne Library, where I communicate, train, and encourage staff to use Nextcloud's Deck. So, I'm adding a +1 to this conversation! Thank you :)

**Comment by Giorgio-Germani** [comment]
yeah this feature would make sense. i think the deck functionality is ok, but quite rudimentary. maybe a look at https://planka.app/ could spark some inspiration to make it a more robust and useful tool. 

time-tracking is certainly the most missed feature still.

**Comment by spielhoelle** [comment]
🆙

**Comment by arongergely** [comment]
Plus one for this feature request from me :)

**Comment by JoeMudr** [comment]
Would like to have this feature too.

**Comment by magrusch** [comment]
I would also appreciate this feature to make it transparent how much effort was spent for deck tasks 👍

Few weeks ago a project was started to implement this: https://github.com/rallisf1/nextcloud-deck-time-tracking

Maybe forces can be joint here 🤝 Just fyi

**Comment by hublex** [comment]
There is already an incredible app (TimeManager). Isn't it possible to combine this two powerful tools. I am using both but think about leaving this tools for a commercial all in one which is optimize for each other.
Creating tickets and or calendar entries is necessary but creating three entries to the same task is a bit too much, especially when doing many different tasks every day.
